Add upgrade pairs to upgrade type

spiderbiggen opened the pull request:
Change `upgrade_type(<ID>, <SourceFunctionID>, <TargetFunctionID>)` to `upgrade_type(<ID>, [upgrade_pair(<SourceFunctionID>, <TargetFunctionID>)...])` to support multiple pairs per upgrade

spiderbiggen committed:
enable multiple upgrade_pairs in upgrade_type

spiderbiggen committed:
fix .* checkstyle error

spiderbiggen committed:
Merge branch 'EnhanceUpgradeTypes' into addUpgradePairsToUpgradeType

# Conflicts:
#	environment/src/main/java/tygronenv/EisEnv.java

klikooo commented on contextvh/src/test/java/contextvh/translators/J2UpgradeTypeTest.java:
```diff
@@ -24,23 +26,24 @@
 	private UpgradeType upgradeType;
 	private UpgradePair upgradePair;
-	private ArrayList<UpgradePair>  a;
+	private List<UpgradePair> a;
```
Maybe change to a more descriptive name?

spiderbiggen replied:
fair point

spiderbiggen commented:
travis buld log is [here](https://travis-ci.org/spiderbiggen/tygron/builds/135965489)

spiderbiggen committed:
Rename variable a to pairs

spiderbiggen commented:
new travis build log is here https://travis-ci.org/spiderbiggen/tygron/builds/135969619

dionytadema commented:
conflicts :(

MHMCD commented:
[![Build Status](https://travis-ci.org/spiderbiggen/tygron.svg?branch=addUpgradePairsToUpgradeType)](https://travis-ci.org/spiderbiggen/tygron)

Easier to check travis build logs

spiderbiggen committed:
Merge remote-tracking branch 'eishub/context' into addUpgradePairsToUpgradeType

# Conflicts:
#	contextvh/src/main/java/contextvh/ContextEnv.java
#	contextvh/src/main/java/contextvh/translators/J2UpgradeType.java

spiderbiggen commented:
fixed merge conflicts

jannelie commented:
we approve -CrashTestDummies

hmyeh commented:
we have looked statically at the code and approve of that - Danshal

MHMCD commented:
We approve

\- CodeFox [![CodeFox](https://avatars1.githubusercontent.com/u/18596181?v=3&s=32)](https://github.com/codefoxnl)

MarcianoJ commented:
Code looks OK, was this also tested with the agent?

spiderbiggen commented:
Marciano yes it was

klikooo commented on contextvh/src/main/java/contextvh/translators/J2UpgradeType.java:
```diff
@@ -21,18 +20,17 @@
+     * {@code upgrade_type(<ID>)} if there is no pair available.
      */
     @Override
     public Parameter[] translate(final UpgradeType u) throws TranslationException {
```
Does checkstyle pass with such a short variable name?

tvleest commented:
This can probably be merged now?

MaxGroenenboom commented:
I think the note @klikooo placed should be addressed.

tvleest commented:
Ah overlooked that one.
According to travis it passed, but it shouldn't? 
The name isn't changed by this PR though, but we could change it.

spiderbiggen commented:
Checkstyle passes, but as @tvleest said we didn't change this it was already that way when i started working on it

klikooo commented on contextvh/src/main/java/contextvh/translators/J2UpgradePair.java:
```diff
@@ -0,0 +1,45 @@
+     * {@code upgrade_type(<ID>)} if there is no pair available.
+     */
+    @Override
+    public Parameter[] translate(final UpgradePair u) throws TranslationException {
```
Change variable name to a more descriptive one

tvleest committed:
Fixed unclear variable names

With these new variable names we will prevent lots of confusion for our fellow students

tvleest commented:
Changed variable names. If there aren't any other concerns, we can merge.

MaxGroenenboom commented:
Looks good, I approve ;)
